src/oas3/accessors.ts

```diff
@@ -1,17 +1,16 @@
 import { get } from 'lodash';
 import { OpenAPIObject, OperationObject, SecuritySchemeObject } from 'openapi3-ts';
 
-import { mapToKeys } from '../utils';
+import { isObject, mapToKeys } from '../utils';
 
 export function getSecurities(
   spec: Partial<OpenAPIObject>,
```

Comment:
Should we change to `DeepPartial` here and in the operation property?

In general using deep partial on entire open api specifications, and then handling all the property checks manually everywhere, is going to succkk. Wondering if there is a better way? Lodash? Although I suppose we lose some typings, but in reality the underlying spec (openapi) is really not changing very often.

Reply:
> Lodash? Although I suppose we lose some typings, but in reality the underlying spec (openapi) is really not changing very often.

Frankly, let's wait for TS 3.7 to be released so that we could utilize optional chaining operator and null null coalescing operator.
Doing so would still let us have fairly clean code, while still preserving typings.

Reply:
Seems like 3.7 beta is [out](https://devblogs.microsoft.com/typescript/announcing-typescript-3-7-beta/) already.
How do you feel about trying it out in http-spec?